packages/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -32,19 +33,34 @@ const mergeProps = makeMergeProps<MenuState>({ deepMerge: menuShorthandProps });
 export const useMenu = (props: MenuProps, ref: React.Ref<HTMLElement>, defaultProps?: MenuProps): MenuState => {
   const { document } = useFluent();
   const triggerId = useId();
+  const isSubmenu = useMenuContext(context => context.hasMenuContext);
 
   const state = mergeProps(
     {
       ref: useMergedRefs(ref, React.useRef(null)),
       menuPopup: { as: 'div' },
-      position: 'below',
-      align: 'start',
+      position: isSubmenu ? 'after' : 'below',
+      align: isSubmenu ? 'top' : 'start',
+      onHover: isSubmenu,
       triggerId,
     },
     defaultProps,
     resolveShorthandProps(props, menuShorthandProps),
   );
 
+  const [checkedValues, setCheckedValues] = useControllableValue(state.checkedValues, state.defaultCheckedValues);
+  state.checkedValues = checkedValues;
+  const { onCheckedValueChange: onCheckedValueChangeOriginal } = state;
+  state.onCheckedValueChange = (e, name, checkedItems) => {
+    if (onCheckedValueChangeOriginal) {
+      onCheckedValueChangeOriginal(e, name, checkedItems);
+    }
+
+    setCheckedValues(s => {
+      return s ? { ...s, [name]: checkedItems } : { [name]: checkedItems };
+    });
+  };
```

Comment:
Shouldn't use `useCallback` in this case?

Reply:
Yes, it should use as it will be passed into a context and then used in selectors:

```
  const onCheckedValueChange = useMenuContext(context => context.onCheckedValueChange);
```

Reply:
used `useEventCallback` since user props can lead to invalidating callbacks quite frequently